src/routes/network/index.ts

```diff
@@ -27,6 +31,152 @@ async function network(fastify: FastifyInstance) {
+                genesisData.active_slots_coefficient,
+              ),
+            };
+            for (let index = 0; index < 3; index++) summary.push(firstLike);
+          }
+
+          // Intermediate summary items separated by protocol major change
+          // excluding inter-era hard forks
+          let previous = first;
+
+          for (const epochProto of protocols.rows) {
+            if (PROTOCOL_VERSIONS[epochProto.protocol_major].is_era_hardfork) {
+              const effectiveEpoch = epochProto.epoch + 1;
+              const duration = effectiveEpoch - previous.end.epoch;
+              const next = {
+                start: {
+                  time: previous.end.time,
+                  slot: previous.end.slot,
+                  epoch: previous.end.epoch,
+                },
+                end: {
+                  time:
+                    previous.end.time +
+                    duration * genesisData.epoch_length * genesisData.slot_length,
+                  slot: previous.end.slot + duration * genesisData.epoch_length,
+                  epoch: effectiveEpoch,
+                },
+                parameters: {
+                  epoch_length: genesisData.epoch_length,
+                  slot_length: genesisData.slot_length,
+                  safe_zone: standardSafeZone(
+                    genesisData.security_param,
+                    genesisData.active_slots_coefficient,
+                  ),
+                },
+              };
+
+              summary.push(next);
+              previous = next;
+            }
+          }
+
+          // Last summary entry is ending at current epoch + 2
+          const lastListedEpoch = lastEpoch.rows[0].epoch + 2;
+          const lastDuration = lastListedEpoch - previous.end.epoch;
+          const last = {
+            start: {
+              time: previous.end.time,
+              slot: previous.end.slot,
+              epoch: previous.end.epoch,
+            },
+            end: {
+              time:
+                previous.end.time +
+                lastDuration * genesisData.epoch_length * genesisData.slot_length,
+              slot: previous.end.slot + lastDuration * genesisData.epoch_length,
+              epoch: lastListedEpoch,
+            },
+            parameters: {
+              epoch_length: genesisData.epoch_length,
+              slot_length: genesisData.slot_length,
+              safe_zone: standardSafeZone(
+                genesisData.security_param,
+                genesisData.active_slots_coefficient,
+              ),
+            },
+          };
+
+          summary.push(last);
+
+          clientDbSync.release();
+          return reply.send(summary);
+        } catch (error) {
+          if (clientDbSync) {
+            clientDbSync.release();
+          }
+          throw error;
+        }
+      } else {
+        return reply.code(500).header('Content-Type', 'application/json; charset=utf-8').send({
```

Comment:
we have shortcut for this

```
return handle500(reply, 'No genesis or Byron data', request);
```

(the string message is used only in log output, it sends generic internal server error)!

Reply:
Thanks!